### PR TITLE
Update Nix backend to use Nix Flakes

### PR DESCRIPTION
**Pull request by matthewbauer**
This is a large overhaul of the Nix backend. It now is much faster
since it can make use of the eval caching that is available in Nix 2.4.

### CODE REVIEW

**Comment by hughsie** [comment]
I have no problem with this, but do we know who is the current "maintainer" of the Nix backend for PK?

**Comment by matthewbauer** [comment]
> I have no problem with this, but do we know who is the current "maintainer" of the Nix backend for PK?

I think it's probably me! I was the last to modify it at least.

**Review by hughsie — APPROVED** [review]
Looks good to me, but please squash before merging

**Pull request merged**
